packages/lab/src/carousel/CarouselContext.tsx

```diff
@@ -0,0 +1,154 @@
+import { createContext, useResizeObserver } from "@salt-ds/core";
+import {
+  type KeyboardEvent,
+  type MouseEvent,
+  type ReactNode,
+  type RefObject,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+
+export interface CarouselContextValue {
+  firstVisibleSlide: number;
+  visibleSlides: number;
+  slidesCount: number;
+  nextSlide: (event: MouseEvent | KeyboardEvent) => void;
+  prevSlide: (event: MouseEvent | KeyboardEvent) => void;
+  updateActiveFromScroll: (scrollLeft: number) => void;
```

Comment:
how do you see Active and Visible?
Are they the same thing. On first read I have to read through code to try and understand the difference and I think they could be the same thing.

Reply:
Any need to have an infinite carousel, so when it gets to the end, it goes back to the beginning?